### DISCUSSION TITLE
Melhores práticas para desenvolvimento mobile focando em economia de dados

### DISCUSSION THREAD

**Discussion by laurindo** [frontendbr/forum#865] [Perguntas e Respostas]
Pessoal, estou criando essa issue porque desejo saber as melhores práticas no desenvolvimento mobile(no meu caso estou usando react-native) para tentar evitar ao máximo o consumo de dados desnecessários em uma aplicação mobile.

Queria saber de vocês quais melhores práticas, ferramentas, links, abordagens, etc para termos uma aplicação mobile que saiba gerenciar essa questão.

Vou ler mais sobre PWA(Progressive Web Apps) que com certeza tem muita coisa legal que fala sobre algumas desses tópicos. 

Mas resumindo, desejo saber de vocês. Desde já agradeço.

**Comment by felipefialho** [comment]
Linkando: #470 #34 #486

**Comment by ninetails** [comment]
Sobre economia de tráfego de dados em relação a comunicação (impactando em mais processamento do servidor), diria para ir atrás de GraphQL, se seu app não for estático (ou seja, há uso de banco de dados).

Já se for economia de tráfego de dados em relação aos assets, no caso do PWA, diria pra dar uma olhada em [Code Splitting](https://webpack.js.org/guides/code-splitting/) (tem artigos no medium que fala melhor sobre técnicas para Code Splitting), Server side rendering (isomorphic/universal rendering) e [Bundle Analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer) para tentar diminuir as dependências do seu bundle + otimização de chunks & cache do navegador.
